### PR TITLE
Add api key authentication mechanism

### PR DESCRIPTION
**Pull request by abnerjacobsen**
Hi,

In the project I'm working on, the API authenticates using API Key in a custom header field.

This pull request adds a new authentication mechanism for these cases.

I take this opportunity to thank you for your excellent work on this project.

Regards

### CODE REVIEW

File: src/apiwrappers/auth.py [hunk 1]
```diff
@@ -31,3 +31,18 @@ def __repr__(self) -> str:
 
     def __call__(self) -> Dict[str, str]:
         return {"Authorization": str(self)}
+
+
+class ApiKeyAuth:
+    def __init__(self, key: str, header: str = "apikey"):
```
Comment (unmade) [inline src/apiwrappers/auth.py]: Can you please make `X-Api-Key` a default value? AFAIK it is a common header name for API Key authentication

File: src/apiwrappers/auth.py [hunk 2]
```diff
@@ -31,3 +31,18 @@ def __repr__(self) -> str:
...
+        self.key = key
+        self.header = header
+
+    def __str__(self) -> str:
+        return f"{self.header} {self.key}"
```
Comment (unmade) [inline src/apiwrappers/auth.py]: Both `BasicAuth` and `TokenAuth` do not contain a header name in `__str__` method. Could you please   provide same behaviour here or at least separate header name and header value with ':'

**Review by unmade — CHANGES_REQUESTED** [review]
Thank you for your contribution! I think it would be useful to have in the library.

I requested some changes, feel free to start a discussion if you have any question or disagree with them

File: src/apiwrappers/auth.py (hunk 1 above)
Comment (abnerjacobsen) [reply src/apiwrappers/auth.py]: modified as requested

File: src/apiwrappers/auth.py (hunk 2 above)
Comment (abnerjacobsen) [reply src/apiwrappers/auth.py]: modified as requested

**Pull request merged**